src/model.js

```diff
@@ -228,8 +231,32 @@ class XTerminalModel {
 		this.element.ptyProcess.write(text)
 	}
 
+	setActive () {
+		if (this.isVisible()) {
+			recalculateActive(this.terminals_set, this)
+		}
+	}
+
+	isVisible () {
+		return this.pane && this.pane.getActiveItem() === this && (!this.dock || this.dock.isVisible())
+	}
+
 	setNewPane (pane) {
 		this.pane = pane
+		const location = this.pane.getContainer().getLocation()
```

Comment:
somewhere is failing due to:

l/pull/23/checks?check_run_id=505027799#step:7:591

Reply:
Ya that test is mocking the `pane` with an empty object. I'm not worried about the tests right now. I'll write tests and fix the ones that are failing once we figure out exactly how this is all going to work.